pkg/e2e/verify/check_workloads.go

```diff
@@ -0,0 +1,133 @@
+package verify
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/openshift/osde2e/pkg/common/alert"
+	"github.com/openshift/osde2e/pkg/common/helper"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	v1 "k8s.io/client-go/kubernetes/typed/core/v1"
+)
+
+var onNodesTestName string = "[Suite: informing] [OSD] pod validating webhook"
+
+func init() {
+	alert.RegisterGinkgoAlert(onNodesTestName, "SD-SREP", "Matt Bargenquast", "sd-cicd-alerts", "sd-cicd@redhat.com", 4)
+}
+
+var _ = ginkgo.Describe(onNodesTestName, func() {
+
+	h := helper.New()
+
+	ginkgo.Context("worker nodes", func() {
+
+		ginkgo.It("on worker nodes", func() {
+
+			_, infra, err := checkWorkloadsOnWorkerNodes(h.Kube().CoreV1())
+			Expect(err).NotTo(HaveOccurred())
+
+			_, err = checkPods(h.Kube().CoreV1(), infra)
+			Expect(err).NotTo(HaveOccurred())
+
+		})
+	})
+})
+
+func checkPods(podClient v1.CoreV1Interface, infra []string) (map[string]string, error) {
+
+	violators := make(map[string]string)
+
+	operators := map[string]string{"cloud-ingress-operator": "openshift-cloud-ingress-operator",
+		"configure-alertmanager-operator": "openshift-monitoring",
+		"custom-domains-operator":         "openshift-custom-domains-operator",
+		"managed-upgrade-operator":        "openshift-managed-upgrade-operator",
```

Comment:
I'll just forewarn that `managed-upgrade-operator` is going to move to master nodes (rather than infra nodes) in OSD-7104 this sprint, so it will soon end up failing this test.

Reply:
@mrbarge thanks for the heads up!